### PR TITLE
Feature/implement wish list

### PR DESCRIPTION
**Pull request by mahmmedn19**

### CODE REVIEW

**Review by Ethaar7 — APPROVED** [review]

File: core/data/src/main/java/org/the_chance/honeymart/data/source/remote/network/HoneyMartService.kt [hunk 1]
```diff
@@ -35,22 +38,22 @@ interface HoneyMartService {
...
+        @Path("id") marketId: Long,
     ): Response<BaseResponse<List<CategoryDto>>>
 
     @POST("/category")
     suspend fun addCategory(
         @Body marketID: Long,
         @Body name: String,
-        @Body imageId: Int
+        @Body imageId: Int,
     ): Response<BaseResponse<CategoryDto>>
```
Comment (AzizaHelmy) [inline core/data/src/main/java/org/the_chance/honeymart/data/source/remote/network/HoneyMartService.kt:49]: Why this changes!

File: user/src/main/java/org/the_chance/honeymart/di/NetworkModule.kt [hunk 2]
```diff
@@ -57,4 +62,12 @@ internal object NetworkModule {
     fun provideHoneyMartService(retrofit: Retrofit): HoneyMartService =
         retrofit.create(HoneyMartService::class.java)
 
+    @Singleton
```
Comment (AzizaHelmy) [inline user/src/main/java/org/the_chance/honeymart/di/NetworkModule.kt:65]: Is this interceptor for authorization? To add token to each request?

**Review by AzizaHelmy — APPROVED** [review]
Great work @Eng-Light , @mahmmedn19
